### PR TITLE
Fixes #21306: FileUpload doesn't in webFragment application

### LINKED ISSUE #21306: FileUpload doesn't in webFragment application.
The file upload component does not work when web-fragment application(Project1) is put inside web application(Project2). The file is not uploaded and ,there are no javascript or server side exceptions thrown.

The projects structure:
Project1:
\src\main\java\com\torqueits\fileupload\Bean.java
\src\main\resources\META-INF\resources\single.xhtml
\src\main\resources\META-INF\web-fragment.xml

Project2: (has dependency to Project 1)
\src\main\webapp\WEB-INF\web.xml
#### Environment
Mojarra 2.2.9
Java 8
All browsers
#### Affected Versions
[4.0_dev]

### PR DESCRIPTION
**Pull request by ht26200**
@glassfishrobot Run CI tests please

### CODE REVIEW

**Comment by ht26200** [comment]
@glassfishrobot Run CI tests please

**Review by shingwaichan — CHANGES_REQUESTED** [review]
We should invoke #scanXmlDefinedClassesIfNecessary instead of having the for loopier jsf.
Also, it would be good to add a test case in appserv-tests/devtests/web

**Comment by ht26200** [comment]
@glassfishrobot Run CI tests please

**Review by shingwaichan — DISMISSED** [review]

File: appserver/tests/appserv-tests/devtests/web/servlet-3.0/webFragmentWithFileUpload/WebTest.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) 1997-2017 Oracle and/or its affiliates. All rights reserved.
```
Comment (yaminikb) [inline appserver/tests/appserv-tests/devtests/web/servlet-3.0/webFragmentWithFileUpload/WebTest.java]: Year should be just 2017

File: appserver/tests/appserv-tests/devtests/web/servlet-3.0/webFragmentWithFileUpload/build.xml [hunk 2]
```diff
@@ -0,0 +1,122 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+<!--
+
+    DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+
+    Copyright (c) 1997-2017 Oracle and/or its affiliates. All rights reserved.
```
Comment (yaminikb) [inline appserver/tests/appserv-tests/devtests/web/servlet-3.0/webFragmentWithFileUpload/build.xml]: Year should be just "2017"

File: appserver/tests/appserv-tests/devtests/web/servlet-3.0/webFragmentWithFileUpload/descriptor/web-fragment.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<web-fragment version="3.0" xmlns="http://java.sun.com/xml/ns/javaee" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd">
```
Comment (yaminikb) [inline appserver/tests/appserv-tests/devtests/web/servlet-3.0/webFragmentWithFileUpload/descriptor/web-fragment.xml]: Missing copyright

File: appserver/tests/appserv-tests/devtests/web/servlet-3.0/webFragmentWithFileUpload/docroot/single.xhtml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yaminikb) [inline appserver/tests/appserv-tests/devtests/web/servlet-3.0/webFragmentWithFileUpload/docroot/single.xhtml]: Missing copyright

File: appserver/tests/appserv-tests/devtests/web/servlet-3.0/webFragmentWithFileUpload/servlet/Bean.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
+package fileupload;
```
Comment (yaminikb) [inline appserver/tests/appserv-tests/devtests/web/servlet-3.0/webFragmentWithFileUpload/servlet/Bean.java]: Missing copyright

**Review by yaminikb — CHANGES_REQUESTED** [review]

**Comment by ht26200** [comment]
Starting CI tests run

**Comment by ht26200** [comment]
@glassfishrobot Run CI tests please

**Comment by ht26200** [comment]
@glassfishrobot Run CI tests please

**Review by yaminikb — APPROVED** [review]

**Pull request merged**